Travel Scheduler/View Controllers/HomeCollectionViewController.m

```diff
@@ -174,22 +161,26 @@ - (NSIndexPath *)tableView:(UITableView *)tableView willSelectRowAtIndexPath:(NS
 
 #pragma mark - UICollectionViewDataSource Methods
 
-- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
-{
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
     return 5;
 }
 
-- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
-{
-    // ----- HEEEEEREEEEE --------
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     [collectionView registerClass:[AttractionCollectionCell class] forCellWithReuseIdentifier:@"AttractionCollectionCell"];
     AttractionCollectionCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"AttractionCollectionCell" forIndexPath:indexPath];
 
     cell.delegate = self;
-    [cell setImage:nil];
+    if (self.areWeInRestaurant) {
+        cell.place = self.hub.dictionaryOfArrayOfPlaces[@"restaurant"][indexPath.row];
```

Comment:
I'm not sure if this is a good suggestion, but I feel like if I were doing this, I wouldn't make three boolean values for areWeInRestaurant, areWeInAttractions... but instead make one NSString *typeOfPlace and set that to @"restaurant", @"attraction"... in the previous if chunk (lines 97-121). This way, your code here would only be one line instead of 3 if statements because you could just say cell.place = self.hub.dictionaryOfArrayOfPlaces[typeOfPlace][indexPath.row]; and it would work for all 3 cases because typeOfPlace is already set to the string value you want.

Reply:
Done!